open-metadata-implementation/view-services/glossary-author-view/glossary-author-view-server/src/main/java/org/odpi/openmetadata/viewservices/glossaryauthor/services/GlossaryAuthorViewGlossaryRESTServices.java

```diff
@@ -361,10 +364,60 @@ public GlossaryAuthorViewGlossaryRESTServices() {
             SubjectAreaNodeClients clients = instanceHandler.getSubjectAreaNodeClients(serverName, userId, methodName);
             Glossary glossary = clients.glossaries().restore(userId, guid);
             response.addResult(glossary);
-        }  catch (Throwable error) {
-            response =  getResponseForError(error, auditLog, className, methodName);
+        } catch (Throwable error) {
+            response = getResponseForError(error, auditLog, className, methodName);
         }
         restCallLogger.logRESTCallReturn(token, response.toString());
         return response;
     }
+    /**
+     * Create the supplied list of Terms in the glossary, identified by the supplied guid. Each term does not need to specify a glossary.
+     *
+     * @param serverName       local UI server name
+     * @param userId           user identifier
+     * @param guid             guid of the glossary under which the Terms will be created
+     * @param terms            terms to create
+     * @return a response which when successful contains a list of the responses from the Term creates (successful or otherwise). The order of the responses is the same as the supplied terms order.
+     *
+     * when not successful the following Exception responses can occur
+     * <ul>
+     * <li> UnrecognizedGUIDException            the supplied guid was not recognised</li>
+     * <li> UserNotAuthorizedException           the requesting user is not authorized to issue this request.</li>
+     * <li> FunctionNotSupportedException        Function not supported</li>
+     * <li> InvalidParameterException            one of the parameters is null or invalid.</li>
+     * <li> MetadataServerUncontactableException not able to communicate with a Metadata respository service.</li>
+     * </ul>
+     */
+    public SubjectAreaOMASAPIResponse<SubjectAreaOMASAPIResponse<Term>> createMultipleTermsInAGlossary(String serverName, String userId, String guid, Term[] terms) {
```

Comment:
Could you make it like this. I would you it like this.

```
    public SubjectAreaOMASAPIResponse<Term> createMultipleTermsInAGlossary(String serverName, String userId, String guid, Term[] terms) {
        final String methodName = "createMultipleTermsInAGlossary";
        RESTCallToken token = restCallLogger.logRESTCall(serverName, userId, methodName);
        SubjectAreaOMASAPIResponse<Term> response = new SubjectAreaOMASAPIResponse<>();
        AuditLog auditLog = null;
        
        try {
            SubjectAreaNodeClients clients = instanceHandler.getSubjectAreaNodeClients(serverName, userId, methodName);
            auditLog = instanceHandler.getAuditLog(userId, serverName, methodName);
            for (Term term : terms) {
                GlossarySummary glossarySummary = new GlossarySummary();
                glossarySummary.setGuid(guid);
                term.setGlossary(glossarySummary);
                Term createdTerm = clients.terms().create(userId, term);
                response.addResult(createdTerm);
            }
        } catch (Throwable error) {
            response = getResponseForError(error, auditLog, className, methodName);
        }
        restCallLogger.logRESTCallReturn(token, response.toString());
        return response;
    }
```

Reply:
In general, if I were you, I would solve this problem on the UI by sending several requests to this method -> `public SubjectAreaOMASAPIResponse<Term> createTerm(String serverName, String userId, Term suppliedTerm)`, since in fact we are doing the same thing here.

Reply:
@immmus thanks for your feedback - I have used your suggested code. In terms of your suggestion to do this in the UI, we have had extensive discussions and the consensus is that we should push this sort of business logic into the view server. This makes it very easy to issue one call in the UI. In general this sort of change is what the View services are for, surfacing UI specific artifacts that can be easily consumed.  I started doing this in the UI, the code would have been quite involved (I looked at solutions involving promise all, a new context and reducer for this, either way there is state to by managed in the Ui code); it is much neater to have it all in this little method.

Reply:
@immmus the new fix is ready for review again.